File: CS/T449148/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using T449148.Models;

namespace T449148.Controllers
{
    public class HomeController : Controller
    {
       [HttpGet]
        public ActionResult Index()
        {
            if (Session["DataTableModel"] == null)
                Session["DataTableModel"] = InMemoryModel.OpenExcelFile("");

            return View(Session["DataTableModel"]);   
        }
        [HttpPost]
        public ActionResult GridViewPartial(string path)
        {
            var model = Session["DataTableModel"];
            if (!string.IsNullOrEmpty(path))
            {
                model = InMemoryModel.OpenExcelFile(path);
                Session["DataTableModel"] = model;
            }
            return PartialView(model);
        }

        public ActionResult UploadControlUpload()
        {
            UploadControlExtension.GetUploadedFiles("UploadControlFile", UploadControlSettings.UploadValidationSettings, UploadControlSettings.FileUploadComplete);
            return null;
        }
    }
    public class UploadControlSettings
    {
        public static DevExpress.Web.UploadControlValidationSettings UploadValidationSettings = new DevExpress.Web.UploadControlValidationSettings()
        {
            AllowedFileExtensions = new string[] {".xlsx" },
            MaxFileSize = 40000000
        };
        public static void FileUploadComplete(object sender, DevExpress.Web.FileUploadCompleteEventArgs e)
        {
            const string UploadDirectory = "~/Content/UploadedFiles/";
            if (e.UploadedFile.IsValid)
            {
                MemoryStream ms = new MemoryStream();
                string resultExtension = Path.GetExtension(e.UploadedFile.FileName);
                string resultFileName = Path.ChangeExtension(Path.GetRandomFileName(), resultExtension);
                string resultFileUrl = UploadDirectory + resultFileName;
                string resultFilePath = System.Web.HttpContext.Current.Server.MapPath("~/") + e.UploadedFile.FileName;
                e.UploadedFile.SaveAs(resultFilePath);
                e.CallbackData = resultFilePath;

            }
        }
    }

}

File: VB/T449148_VB/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports System.Web.Mvc
Imports System.IO

Namespace Controllers
    Public Class HomeController
        Inherits Controller

        ' GET: Home
        Function Index() As ActionResult
            If (Session("DataTableModel") Is Nothing) Then
                Session("DataTableModel") = InMemoryModel.OpenExcelFile("")
            End If
            Return View(Session("DataTableModel"))

        End Function

        Public Function GridViewPartial(ByVal path As String) As ActionResult
            Dim model = Session("DataTableModel")
            If Not String.IsNullOrEmpty(path) Then
                model = InMemoryModel.OpenExcelFile(path)
                Session("DataTableModel") = model
            End If
            Return PartialView(model)
        End Function
        Public Function UploadControlUpload() As ActionResult
            UploadControlExtension.GetUploadedFiles("UploadControlFile", UploadControlSettings.UploadValidationSettings, AddressOf UploadControlSettings.FileUploadComplete)
            Return Nothing
        End Function

    End Class

    Public Class UploadControlSettings
        Public Shared UploadValidationSettings As DevExpress.Web.UploadControlValidationSettings = New DevExpress.Web.UploadControlValidationSettings() With {.AllowedFileExtensions = New String() {".xlsx"}, .MaxFileSize = 40000000}

        Public Shared Sub FileUploadComplete(ByVal sender As Object, ByVal e As DevExpress.Web.FileUploadCompleteEventArgs)
            Const UploadDirectory As String = "~/Content/UploadedFiles/"
            If e.UploadedFile.IsValid Then
                Dim ms As MemoryStream = New MemoryStream()
                Dim resultExtension As String = Path.GetExtension(e.UploadedFile.FileName)
                Dim resultFileName As String = Path.ChangeExtension(Path.GetRandomFileName(), resultExtension)
                Dim resultFileUrl As String = UploadDirectory & resultFileName
                Dim resultFilePath As String = System.Web.HttpContext.Current.Server.MapPath("~/") + e.UploadedFile.FileName
                e.UploadedFile.SaveAs(resultFilePath)
                e.CallbackData = resultFilePath
            End If
        End Sub
    End Class
End Namespace